### ISSUE TITLE
LOWEST_VAL forgets the historical best after resume: _best_val_loss is never restored

### ISSUE THREAD

**Issue by AlfredQin** [NVIDIA-NeMo/Automodel#3040]
### Describe the bug

`BaseRecipe` tracks the best validation loss behind the `LOWEST_VAL` checkpoint symlink in an in-memory attribute that is never checkpointed or restored:

```python
# nemo_automodel/recipes/base_recipe.py
if "_best_val_loss" not in self.__dict__:
    self.__dict__["_best_val_loss"] = float("inf")
...
def _update_best_symlink(self, target_dir: str, val_loss: float) -> None:
    ...
    if val_loss < self._best_val_loss:
        self._best_val_loss = val_loss
        self._update_checkpoint_symlink("LOWEST_VAL", target_dir)
```

`load_checkpoint` restores model/optimizer/scheduler/step state but not `_best_val_loss`, so after any resume the attribute is `inf` again and the **first** post-resume validation checkpoint always replaces `LOWEST_VAL` — even when its loss is worse than the pre-resume best. The symlink's documented contract ("the checkpoint with the lowest validation loss") is broken across restarts, which is exactly when long runs rely on it.

Observed at `14d177313a744483ce5410e812fb3736adc8846c`; the relevant code is unchanged on `main` at `e4e9efd`.

### Steps/Code to reproduce bug

1. Train with a validation loop until a checkpoint with val_loss `A` becomes `LOWEST_VAL`.
2. Interrupt and resume from `LATEST`.
3. Let the next validation produce val_loss `B > A`.
4. `LOWEST_VAL` now points at the `B` checkpoint.

### Expected behavior

Persist the best value in the tracked recipe state (save/restore it with the checkpoint), or reconstruct it on load from the existing `LOWEST_VAL` target's recorded loss before making new symlink decisions.

**Comment by HuiyingLi** [comment]
Thanks for the issue! Looks like we have a related PR #2416  that might fix this.

**Comment by AlfredQin** [comment]
Thanks for your reply.

**Comment by HuiyingLi** [comment]
Keeping it open since 2416 is not merged yet.
